src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -450,12 +437,6 @@ private void ParseArray(ParseRecord pr)
                     }
                 }
 
-                // For binary, headers comes in as an array of header objects
-                if (pr._objectPositionEnum == InternalObjectPositionE.Headers)
```

Comment:
https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryParser.cs#L775 uses

I think we should let it be there and not start throwing in the parser